catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java

```diff
@@ -817,6 +817,16 @@ public boolean login(String username, String password, String token, Context con
 			String serverAnswer = jsonObject.optString(JSON_ANSWER);
 			SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
 
+			if (BuildConfig.CREATE_AT_SCHOOL) {
+				boolean createAtSchoolUser = jsonObject.optBoolean(Constants.CREATE_AT_SCHOOL_USER);
+				sharedPreferences.edit().putBoolean(Constants.CREATE_AT_SCHOOL_USER, createAtSchoolUser).apply();
+				if (!createAtSchoolUser) {
+					sharedPreferences.edit().putString(Constants.TOKEN, Constants.NO_TOKEN).apply();
```

Comment:
Yes, now I remember the reason --> In the create@school-flavor there is an additional Tracking done in Utils.logoutUser() -- TrackingUtil.trackLogoutEndSessionEvent(context);
This should not be logged in the case of an error here.
Facebook and Google+ logout isn't relevant, as only native login is provided in this flavor. Furthermore, we don't want to show the Toast message "Logout successful". So it should be perfectly fine here to reset those 2 entries in the shared prefs.